src/Command/UnregistrerAdherentsCommand.php

```diff
@@ -0,0 +1,175 @@
+                $output->writeln($email);
+            }
+        }
+        if (0 < count($this->hosts)) {
+            $output->writeln('The adherents with following email adress are hosts of committees. They are not unregistrated :');
+            foreach ($this->hosts as $email) {
+                $output->writeln($email);
+            }
+        }
+        if (0 < count($this->citizenProjectAdministrators)) {
+            $output->writeln('The adherents with following email adress are administrators of citizen projects. They are not unregistrated :');
+            foreach ($this->citizenProjectAdministrators as $email) {
+                $output->writeln($email);
+            }
+        }
+    }
+
+    private function parseCSV(string $filename): array
+    {
+        $emails = [];
+        if (false === ($handle = fopen($filename, 'r'))) {
+            throw new FileNotFoundException(sprintf('% not found', $filename));
+        }
+
+        $isFirstRow = true;
+        while (false !== ($data = fgetcsv($handle, 100000, ';'))) {
+            if (true === $isFirstRow) {
+                $isFirstRow = false;
+
+                continue;
+            }
+
+            $row = array_map('trim', $data);
+            $emails[] = $row[4];
+        }
+        fclose($handle);
+
+        return $emails;
+    }
+
+    private function unregistrateAdherents(array $emails, OutputInterface $output): bool
+    {
+        $batchSize = 20;
+        $i = 1;
+        $progress = new ProgressBar($output, count($emails));
+        $progress->start();
+
+        foreach ($emails as $email) {
+            try {
+                if (null === ($adherent = $this->adherentRepository->findByEmail($email))) {
+                    $this->emailNotFound[] = $email;
+
+                    continue;
+                }
+
+                if ($adherent->isHost()) {
+                    $this->hosts[] = $email;
+
+                    continue;
+                }
+
+                if ($adherent->isAdministrator()) {
+                    $this->citizenProjectAdministrators[] = $email;
+
+                    continue;
+                }
+
+                $this->adherentManager->unregister($adherent, false);
+            } catch (\Exception $ex) {
+                $output->writeln(
+                    sprintf('An error occured while unregistering an adherent with email %s : %s', $email, $ex->getMessage())
+                );
+
+                return true;
+            }
+
+            if (0 === ($i % $batchSize)) {
+                $progress->advance($batchSize);
+                $progress->display();
+                $this->em->flush();
+                $this->em->clear(); // Detaches all objects from Doctrine for memory save
+            }
+
+            ++$i;
+        }
+
+        $this->em->flush(); //Persist objects that did not make up an entire batch
+        $this->em->clear();
+
+        $progress->finish();
+
+        return false;
```

Comment:
Returning false to say that everything went ok is a bit counter intuitive. This private method should not return a boolean but throw a runtime exception instead of returning true.

Reply:
Done